qiskit/backends/baseprovider.py

```diff
@@ -52,9 +54,9 @@ def get_backend(self, name=None, **kwargs):
         """
         backends = self.backends(name, **kwargs)
         if len(backends) > 1:
-            raise KeyError('More than one backend matches the criteria')
+            raise QiskitBackendNotFoundError('More than one backend matches the criteria')
         elif not backends:
-            raise KeyError('No backend matches the criteria')
+            raise QiskitBackendNotFoundError('No backend matches the criteria')
```

Comment:
Can you update the docstring for this function, updating the `Raises` description to match the new exception being thrown, as this file acts as the "public api" of sorts?

Reply:
Done.